### PR TITLE
Adding e2e test for update and delete of resourcequota

### LINKED ISSUE #74569: Adding e2e test for update and delete of resourcequota
**What would you like to be added**:

e2e test suites for:
- PUT /api/v1/namespaces/{namespace}/resourcequotas/{name}
- DELETE /api/v1/namespaces/{namespace}/resourcequotas/{name}

**Why is this needed**:

The above APIs are not performed in current e2e test. By adding test for it, e2e test coverage is improved.

/sig testing

### PR DESCRIPTION
**Pull request by s-ito-ts**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This patch adds e2e test for update and delete of resourcequota to cover the following API.
 -  PUT /api/v1/namespaces/{namespace}/resourcequotas/{name}
 -  DELETE /api/v1/namespaces/{namespace}/resourcequotas/{name}

**Which issue(s) this PR fixes**:

Fixes #74569

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/apimachinery/resource_quota.go [hunk 1]
```diff
@@ -664,7 +664,47 @@ var _ = SIGDescribe("ResourceQuota", func() {
...
+		Expect(resourceQuotaResult.Spec.Hard[v1.ResourceCPU]).To(Equal(resource.MustParse("2")))
+		Expect(resourceQuotaResult.Spec.Hard[v1.ResourceMemory]).To(Equal(resource.MustParse("1Gi")))
+
+		By("Verifying a ResourceQuota was modified")
+		resourceQuotaResult, err = client.CoreV1().ResourceQuotas(ns).Get(quotaName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resourceQuotaResult.Spec.Hard[v1.ResourceCPU]).To(Equal(resource.MustParse("2")))
+		Expect(resourceQuotaResult.Spec.Hard[v1.ResourceMemory]).To(Equal(resource.MustParse("1Gi")))
+
+		By("Deleting a ResourceQuota")
```
Comment (mourya007) [inline test/e2e/apimachinery/resource_quota.go]: @liggitt  Hi, just a query that can we ensure the deleted resourceQuota like we are using for secrets, configmap etc. ? For example 		By("Ensuring resource quota status released usage")...

Comment (liggitt) [reply test/e2e/apimachinery/resource_quota.go]: I'm not sure I understand what this is asking

File: test/e2e/apimachinery/resource_quota.go [hunk 2]
```diff
@@ -664,7 +664,47 @@ var _ = SIGDescribe("ResourceQuota", func() {
...
+		err = deleteResourceQuota(client, ns, quotaName)
+		Expect(err).NotTo(HaveOccurred())
```
Comment (oomichi) [inline test/e2e/apimachinery/resource_quota.go]: I guess we can verify the ResourceQuota is deleted by adding the following:
```
By("Verifying the deleted ResourceQuota")
_, err := client.CoreV1().ResourceQuotas(ns).Get(quotaName, metav1.GetOptions{})
Expect(errors.IsNotFound(err)).To(Equal(true))
```

**Review by oomichi — COMMENTED** [review]
Most part is good for me, just one comment.

File: test/e2e/apimachinery/resource_quota.go (hunk 2 above)
Comment (s-ito-ts) [reply test/e2e/apimachinery/resource_quota.go]: Thanks for the review.
I added it.

**Comment by oomichi** [comment]
Thanks for updating,

/lgtm

**Pull request merged**